lib/client.js

```diff
@@ -112,34 +152,60 @@ module.exports = function (dependencies) {
 
     request.write(notification.body);
 
-    return new Promise ( resolve => {
+    return new Promise ( resolve => {      
       request.on("end", () => {
+        try {
+          if (logger.enabled) {
+            logger(`Request ended with status ${status} and responseData: ${responseData}`);
```

Comment:
This results in the error message `apn Request ended with status null and responseData: ` when a request times out or is aborted, which may confuse people reading the logs.

Setting `status = "(timed out)"` in the 'timeout' handler (and possibly something similar for aborted, etc) is one way this error message could be made more readable, but not a blocker

Reply:
This is what I've seen in the actual logs for push attempts to APNs

This can also be reproduced by running `DEBUG=apn node_modules/.bin/mocha test/client.js`

```
  apn Request closed +1ms
  apn Called client.shutdown() +0ms
  apn Session closed +1ms
  apn Session connected +3ms
  apn Request timeout +502ms
  apn Request ended with status null and responseData:  +3ms
  apn Request closed +0ms
  apn Request timeout +501ms
  apn Request timeout +0ms
  apn Request ended with status null and responseData:  +1ms
  apn Request closed +0ms
  apn Request timeout +0ms
  apn Request ended with status null and responseData:  +0ms
  apn Request closed +1ms
  apn Request timeout +0ms
  apn Request ended with status null and responseData:  +0ms
  apn Request closed +0ms
    ✓ Handles APNs timeouts (1011ms)

```

Reply:
ok. just sent an additional commit. could you please check?